### PR TITLE
[JIMT][ADD AMPLITUDE EVENT] - adds a close event

### PR DESCRIPTION
**Pull request by thomasoniii**
We have a discrepancy in the event logging that we're trying to track down, and speculation is that the users may be closing out of the modal w/o clicking the generate button, which indicates a usability issue.

But we're not tracking when the user just closes the modal. So this just adds a new amplitude event attached to the `X` in the modal. Also sends along the selected emojis as well as the modal's current mode.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by cat5inthecradle — APPROVED** [review]

**Review by jmkulwik — APPROVED** [review]

File: apps/src/dance/ai/DanceAiModal.tsx [hunk 1]
```diff
@@ -504,6 +504,15 @@ const DanceAiModal: React.FunctionComponent<DanceAiModalProps> = ({
...
   };
 
+  const handleOnClose = () => {
+    analyticsReporter.sendEvent(EVENTS.DANCE_PARTY_AI_MODAL_CLOSED, {
+      emojis: inputs,
+      mode,
+    });
+
+    onClose();
+  };
```
Comment (breville) [inline apps/src/dance/ai/DanceAiModal.tsx:514]: Does this also cover the case if the user presses the Esc key?

**Comment by cat5inthecradle** [comment]
waiting on drone, unit passed, ui crashed, restarted
https://drone.cdn-code.org/code-dot-org/code-dot-org/37004

File: apps/src/dance/ai/DanceAiModal.tsx [hunk 2]
```diff
@@ -504,6 +504,15 @@ const DanceAiModal: React.FunctionComponent<DanceAiModalProps> = ({
     }
   };
 
+  const handleOnClose = () => {
+    analyticsReporter.sendEvent(EVENTS.DANCE_PARTY_AI_MODAL_CLOSED, {
+      emojis: inputs,
+      mode,
+    });
```
Comment (breville) [inline apps/src/dance/ai/DanceAiModal.tsx:511]: Not crucial, but if we are in the generating mode, we could also record `generatingProgress.step` so we know exactly when it's being closed.

**Comment by cat5inthecradle** [comment]
UI Tests crashed due to unrelated issues, I'm fine merging this if unit tests pass.

**Comment by wilkie** [comment]
I agree. They did pass.

**Pull request merged**